jax/_src/lax/eigh.py

```diff
@@ -62,7 +62,7 @@ def _mask(x, dims, alternative=0):
   mask = None
   for i, d in enumerate(dims):
     if d is not None:
-      mask_dim_i = lax.broadcasted_iota(jnp.int32, x.shape, i) < d
```

Comment:
Hmm. I don't love this on the groups that "stringly-typed" things are lightly type checked.

Use np.int32 instead?

Reply:
Done, here and elsewhere